asv_bench/benchmarks/frame_methods.py

```diff
@@ -862,4 +862,28 @@ def time_last_valid_index(self, dtype):
         self.df.last_valid_index()
 
 
+class Update:
+    def setup(self):
+        rng = np.random.default_rng()
+        self.df = DataFrame(rng.uniform(size=(100_000, 10)))
+
+        idx = rng.choice(range(100_000), size=100_000, replace=False)
+        self.df_random = DataFrame(self.df, index=idx)
+
+        idx = rng.choice(range(100_000), size=10_000, replace=False)
+        cols = rng.choice(range(10), size=2, replace=False)
+        self.df_sample = DataFrame(
+            rng.uniform(size=(10_000, 2)), index=idx, columns=cols
+        )
```

Comment:
```suggestion
        rng = np.random.default_rng()
        self.df = DataFrame(rng.uniform(size=(1_000_000, 10)))

        idx = rng.choice(range(10_000_000), size=1_000_000, replace=False)
        self.df_random = DataFrame(self.df, index=idx)

        idx = rng.choice(range(1_000_000), size=100_000, replace=False)
        cols = rng.choice(range(10), size=2, replace=False)
        self.df_sample = DataFrame(
            rng.uniform(size=(100_000, 2)), index=idx, columns=cols
        )
```

Can you go one more order of magnitude larger. You increased by two orders of magnitude and saw very little change in runtime - meaning that you were mostly measuring overhead. With the above changes, I'm seeing:

```
%timeit df.update(df_sample)
%timeit df_random.update(df_sample)
%timeit df_sample.update(df)
34.7 ms ± 324 µs per loop (mean ± std. dev. of 7 runs, 10 loops each)
17.3 ms ± 158 µs per loop (mean ± std. dev. of 7 runs, 100 loops each)
19.4 ms ± 123 µs per loop (mean ± std. dev. of 7 runs, 100 loops each)
```

which is more in the realm of what we want for ASVs.

Reply:
Done.

Just for consistency, the benchmarks comparing the last version of DataFrame.Update() and the current one. At some point there is some performance crossover, for a smaller dataframe, it seems to be a win-win. But for larger, there is performance decrease in some cases. 

| Change   | Before [4fe49b16] <main>   | After [93ea131f] <fix_gh55509_v2~14>   |   Ratio | Benchmark (Parameter)                                   |
|----------|----------------------------|----------------------------------------|---------|---------------------------------------------------------|
| +        | 40.4±1ms                   | 50.7±0.9ms                             |    1.25 | frame_methods.Update.time_to_update_big_frame_small_arg |
| -        | 68.1±0.8ms                 | 35.9±0.9ms                             |    0.53 | frame_methods.Update.time_to_update_random_indices      |
| -        | 26.4±0.2ms                 | 5.98±0.07ms                            |    0.23 | frame_methods.Update.time_to_update_small_frame_big_arg |